js_13/js/script.js

```diff
@@ -2,76 +2,68 @@
 
 $(function(){
 
-  if (localStorage.getItem('sectionQA')) {
+  if ( !localStorage.getItem('questions') ) { return; }
 
-    var sectionQA = localStorage.getItem('sectionQA');
-
-    var html = $('#content-template').html();
-
-    var tmpl = _.template(html);
-    var content = tmpl(sectionQA);
+  var questions = JSON.parse( localStorage.getItem('questions') );
+  var html = $('#content-template').html();
+  var tmpl = _.template(html);
+  var content = tmpl(questions);
     
-    $('.content').append(content);
+  $('.content').append(content);
     
-    $('.btn').on('click', function(){
-      var correctAnswers = JSON.parse(localStorage.getItem('correctAnswers'));
-      var n = 3;
-      var sectionNumber = -1;
-      var correctAnswer = -1;
-      var results = {
-        id: [ 'вы не отметили', 'вы не отметили', 'вы не отметили' ]
-      };
-
-      $( '.answers li input' ).each(function( index ) {
-        if ($(this).prop('checked')) {
-          sectionNumber = parseInt(index / n);
-          correctAnswer = index - (sectionNumber * n);
-          if (correctAnswers.answerId[sectionNumber] === correctAnswer) {
-            results.id[sectionNumber] = 'верно';
-          } else {
-            results.id[sectionNumber] = 'не верно';
-          }
-        }
-      });
-
-      showModal(results);
-
-      $('.result-modal .btn-close').on('click', function () {
-        hideModal();
-      });
+  $('.btn').on('click', function() {
+    showModal( defineResults('.answers li input') );
 
+    $('.result-modal .btn-close').on('click', function () {
+      hideModal();
     });
 
+  });
+    
+  function defineResults(listElementSelector) {
+    var sectionNumber = -1;
+    var answerIndex = -1;
+    var resultObj = {
+      answer: [ 'вы не отметили', 'вы не отметили', 'вы не отметили' ]
+    };
+  
+    $( listElementSelector ).each(function( liIndex ) {
+      if ( $(this).prop('checked') ) {
+        sectionNumber = parseInt(liIndex / questions.length);
+        answerIndex = liIndex - (sectionNumber * questions.length);
+  
+        if ( questions[sectionNumber].answers[answerIndex].isCorrect ) {
+          resultObj.answer[sectionNumber] = 'верно';
+        } else {
+          resultObj.answer[sectionNumber] = 'не верно';
+        }
```

Comment:
also please not that such simple `if/else` statements can be rewritten with use of ternary operator
```javascript
resultObj.answer[sectionNumber] = questions[sectionNumber].answers[answerIndex].isCorrect ? 'верно' : 'не верно';
```
or
```javascript
resultObj.answer[sectionNumber] = questions[sectionNumber].answers[answerIndex].isCorrect
  ? 'верно'
  : 'не верно';
```

Reply:
Got it